### PR TITLE
instrumentation test for route refresh

### PR DESCRIPTION
**Pull request by VysotskiVadim**
### Description
Follow up PR to address [the comment ](https://github.com/mapbox/mapbox-navigation-android/pull/5767#issuecomment-1141296618)

### CODE REVIEW

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/TestUtils.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+}
+
+suspend fun MapboxNavigation.waitForNewRoute() {
+    waitForRoutesUpdate(RoutesExtra.ROUTES_UPDATE_REASON_NEW)
+}
+
+suspend fun MapboxNavigation.waitForRoutesCleanup() {
+    waitForRoutesUpdate(RoutesExtra.ROUTES_UPDATE_REASON_NEW)
```
Comment (LukasPaczos) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/TestUtils.kt:48]:
```suggestion
    waitForRoutesUpdate(RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP)
```

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt [hunk 2]
```diff
@@ -187,6 +195,44 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
...
+            .getSuccessfulResultOrThrowException()
+            .routes
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        mapboxNavigation.startTripSession()
+        stayOnInitialPosition()
+
+        waitForRouteToRefresh()
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf())
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        waitForRouteToRefresh()
```
Comment (RingerJK) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:211]: can also check that refreshed route has expected annotations, incidents and id?

Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:211]: The goal of this test is to verify that route refreshes doesn't stop working after cleaning up and setting a new route. The way separate routes is refreshed is well tested in [unit tests](https://github.com/mapbox/mapbox-navigation-android/blob/6413f57e6ddd9138ee6e51cca5dd9860d1785435/libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt).

Do you see any aspect of behaviour which you can break without making at least 1 test red?

**Review by LukasPaczos — COMMENTED** [review]
@VysotskiVadim I'm missing a test which waits for the congestion/annotations to expiry, cleans them up, but then is able to provide them again when we eventually succeed.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt (hunk 2 above)
Comment (RingerJK) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:211]: we actually verify it in other tests, so can skip there. Could you rename the test to align the name and aim

**Comment by VysotskiVadim** [comment]
> @VysotskiVadim I'm missing a test which waits for the congestion/annotations to expiry, cleans them up, but then is able to provide them again when we eventually succeed.

@LukasPaczos , see `routeSuccessfullyRefreshesAfterInvalidationOfExpiringData`

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt (hunk 2 above)
Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt]: Does it look better now?

**Review by RingerJK — APPROVED** [review]
LGTM

**Pull request merged**